File: BlackJackJava/Bet.java
public class Bet {
    private int playerBalance = 0;
    private int currentAmount = 0;

    public Bet(int initialBalance){
        this.playerBalance= initialBalance;
        this.currentAmount = 0;
    }

    public void placeBet(int amount){
        if(amount > playerBalance){
            throw new IllegalAccessError("No money");
        }
        currentAmount = amount;
        playerBalance = playerBalance - amount;
    }

    public void winBet(){
        playerBalance = playerBalance + (2 * currentAmount);
        currentAmount = 0;
    }

    public void loseBet(){
        currentAmount = 0;
    } 

    public void tieBet(){
        playerBalance = playerBalance + currentAmount;
        currentAmount = 0;
    }

    public int getBalance(){
        return this.playerBalance;
    }

    public int getCurrentAmount(){
        return this.currentAmount;
    }

}

File: BlackJavaScript/Bet.js
class Bet{
    constructor(initialBalance){
        this.playerBalance = initialBalance;
        this.currentAmount = 0;
    }

    placeBet(amount){
        if(amount > this.playerBalance){
            throw new Error("No money");
        }
        this.currentAmount = amount;
        this.playerBalance -= amount;
    }

    winBet(){
        this.playerBalance += (2 * this.currentAmount);
        this.currentAmount = 0;
    }

    lostBet(){
        this.currentAmount = 0;
    }
    
    tieBet(){
        this.playerBalance += this.currentAmount;
        this.currentAmount = 0;
    }

    getBalance(){
        return this.playerBalance;
    }

    getCurrentAmount(){
        return this.currentAmount;
    }

}